Fix: build issues

Fix these build errors:

```
$ tsc --build
node_modules/@portkey-ai/gateway/src/providers/palm/chatComplete.ts:67:16 - error TS2352: Conversion of type '{ error: { message: string; type: null; param: null; code: number; }; provider: string; }' to type 'ErrorResponse' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  The types of 'error.code' are incompatible between these types.
    Type 'number' is not comparable to type 'string'.

 67         return {
                   ~
 68             error: {
    ~~~~~~~~~~~~~~~~~~~~
...
 74             provider: "palm"
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 75         } as ErrorResponse;
    ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@portkey-ai/gateway/src/providers/palm/complete.ts:62:16 - error TS2352: Conversion of type '{ error: { message: string; type: null; param: null; code: number; }; provider: string; }' to type 'ErrorResponse' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  The types of 'error.code' are incompatible between these types.
    Type 'number' is not comparable to type 'string'.

 62         return {
                   ~
 63             error: {
    ~~~~~~~~~~~~~~~~~~~~
...
 69             provider: "palm"
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 70         } as ErrorResponse;
    ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@portkey-ai/gateway/src/utils.ts:40:7 - error TS2322: Type 'Record<string, any>' is not assignable to type 'T'.
  'T' could be instantiated with an arbitrary type which could be unrelated to 'Record<string, any>'.

40       return obj; // Return unchanged for non-objects or null
         ~~~~~~

node_modules/@portkey-ai/gateway/src/handlers/handlerUtils.ts:205:45 - error TS2345: Argument of type 'RetrySettings' is not assignable to parameter of type 'number'.

205   } else if (providerOption.retry && !isNaN(providerOption.retry)) {
                                                ~~~~~~~~~~~~~~~~~~~~

node_modules/@portkey-ai/gateway/src/handlers/handlerUtils.ts:207:7 - error TS2322: Type 'RetrySettings' is not assignable to type 'number'.

207       attempts: providerOption.retry,
          ~~~~~~~~

  node_modules/@portkey-ai/gateway/src/types/requestBody.ts:7:3
    7   attempts: number;
        ~~~~~~~~
    The expected type comes from property 'attempts' which is declared here on type 'RetrySettings'


Found 5 errors.

error: script "build" exited with code 1 (SIGHUP)
```

--- a/package.json
+++ b/package.json
@@ -43,7 +43,7 @@
     "@rollup/plugin-terser": "^0.4.4",
     "@rollup/plugin-typescript": "^11.1.5",
     "@types/async-retry": "^1.4.5",
-    "@types/node": "^20.10.5",
+    "@types/node": "20.8.3",
     "prettier": "^3.1.1",
     "rollup": "^4.9.1",
     "tsx": "^4.7.0",

--- a/src/handlers/handlerUtils.ts
+++ b/src/handlers/handlerUtils.ts
@@ -202,7 +202,7 @@ export async function tryPostProxy(c: Context, providerOption:Options, inputPara
       attempts: providerOption.retry?.attempts ?? 1, 
       onStatusCodes: providerOption.retry?.onStatusCodes ?? RETRY_STATUS_CODES
     };
-  } else if (providerOption.retry && !isNaN(providerOption.retry)) {
+  } else if (typeof providerOption.retry === "number") {
     providerOption.retry = { 
       attempts: providerOption.retry, 
       onStatusCodes: RETRY_STATUS_CODES

--- a/src/providers/palm/chatComplete.ts
+++ b/src/providers/palm/chatComplete.ts
@@ -1,3 +1,4 @@
+import { PALM } from "../../globals";
 import { Params } from "../../types/requestBody";
 import { PalmChatCompleteResponse } from "../../types/responseBody";
 import { ChatCompletionResponse, ErrorResponse, ProviderConfig } from "../types";
@@ -69,9 +70,9 @@ export const PalmChatCompleteResponseTransform: (response: PalmChatCompleteRespo
                 message: response.error?.message ?? null,
                 type: null,
                 param: null,
-                code: response.error?.code ?? null
+                code: response.error?.code?.toString() ?? null
             },
-            provider: "palm"
+            provider: PALM
         } as ErrorResponse;
     }
 
@@ -80,7 +81,7 @@ export const PalmChatCompleteResponseTransform: (response: PalmChatCompleteRespo
         object: "chat_completion",
         created: Math.floor(Date.now() / 1000),
         model: "Unknown",
-        provider: "palm",
+        provider: PALM,
         choices: response.candidates?.map((generation, index) => ({
             message: { role: "assistant", content: generation.content },
             index: index,

--- a/src/providers/palm/complete.ts
+++ b/src/providers/palm/complete.ts
@@ -1,3 +1,4 @@
+import { PALM } from "../../globals";
 import { Params } from "../../types/requestBody";
 import { PalmCompleteResponse } from "../../types/responseBody";
 import { CompletionResponse, ErrorResponse, ProviderConfig } from "../types";
@@ -64,9 +65,9 @@ export const PalmCompleteResponseTransform: (response: PalmCompleteResponse, res
                 message: response.error?.message ?? null,
                 type: null,
                 param: null,
-                code: response.error?.code ?? null
+                code: response.error?.code?.toString() ?? null
             },
-            provider: "palm"
+            provider: PALM
         } as ErrorResponse;
     }
 
@@ -75,7 +76,7 @@ export const PalmCompleteResponseTransform: (response: PalmCompleteResponse, res
         object: "completion",
         created: Math.floor(Date.now() / 1000),
         model: "Unknown",
-        provider: "palm",
+        provider: PALM,
         choices: response.candidates?.map((generation, index) => ({
             text: generation.output,
             index: index,

--- a/src/utils.ts
+++ b/src/utils.ts
@@ -32,17 +32,17 @@ export const getStreamingMode = (reqBody: Params, provider: string, requestUrl: 
     return reqBody.stream 
 }
 
-export function convertKeysToCamelCase<T>(
+export function convertKeysToCamelCase(
     obj: Record<string, any>,
     parentKeysToPreserve: string[] = []
-  ): T {
+  ): Record<string, any> {
     if (typeof obj !== 'object' || obj === null) {
       return obj; // Return unchanged for non-objects or null
     }
   
     if (Array.isArray(obj)) {
       // If it's an array, recursively convert each element
-      return obj.map((item) => convertKeysToCamelCase(item, parentKeysToPreserve)) as T;
+      return obj.map((item) => convertKeysToCamelCase(item, parentKeysToPreserve));
     }
   
     return Object.keys(obj).reduce((result: any, key: string) => {
@@ -58,7 +58,7 @@ export function convertKeysToCamelCase<T>(
       }
   
       return result;
-    }, {} as T);
+    }, {});
   
     function toCamelCase(snakeCase: string): string {
       return snakeCase.replace(/(_\w)/g, (match) => match[1].toUpperCase());

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,7 +1,6 @@
 {
   "compilerOptions": {
     "strict": true,
-    "strictNullChecks": false,
     "target": "ESNext",
     "module": "ESNext",
     "moduleResolution": "node",

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,11 +1,12 @@
 {
   "compilerOptions": {
+    "strict": true,
+    "strictNullChecks": false,
     "target": "ESNext",
     "module": "ESNext",
     "moduleResolution": "node",
     "esModuleInterop": true,
     "skipLibCheck": true,
-    "strict": true,
     "lib": ["esnext"],
     "types": ["@cloudflare/workers-types", "node"],
     "jsx": "react-jsx",

